lxd/storage_lvm.go

```diff
@@ -238,23 +238,34 @@ func (s *storageLvm) StoragePoolInit(config map[string]interface{}) (storage, er
 	}
 
 	source := s.pool.Config["source"]
-	s.vgName = s.pool.Config["lvm.vg_name"]
 	s.thinPoolName = s.pool.Config["lvm.thinpool_name"]
+	if s.thinPoolName == "" {
+		// Should actually always be set. But let's be sure.
+		s.thinPoolName = "LXDThinpool"
+	}
+	s.thinPoolName = s.pool.Config["lvm.thinpool_name"]
```

Comment:
That makes not sense :)

You're setting s.thinPoolName from the Config, then check if it's empty, if it is, you set it with the default value, which you then overwrite with the same empty value from the Config